File: models/ClothingProduct.dart
import 'Product.dart';

class ClothingProduct extends Product {
  final String size;

  const ClothingProduct({
    required super.id,
    required super.name,
    required super.price,
    required super.category,
    required super.desc,
    required super.imageURL,
    required this.size,
  });

  //converts json
  factory ClothingProduct.fromJson(Map<String, dynamic> json) {
    return ClothingProduct(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      price: (json['price'] as num).toDouble(),
      category: json['category'] ?? '',
      desc: json['desc'] ?? '',
      imageURL: json['imageURL'] ?? '',
      size: json['size'] ?? '',
    );
  }
}

File: INFS4102_project-main/Backend/eshop/src/main/java/qa/udst/eshop/models/ClothingProduct.java
package qa.udst.eshop.models;

import jakarta.validation.constraints.*;

//Clothing: Size
public class ClothingProduct extends Product{
    @NotBlank
    private String size;

    public ClothingProduct(String name, double price, ProductCategory category, String desc, @NotBlank String size, @NotBlank String imageURL) {
        super(name, price, category, desc, imageURL);
        this.size = size;
    }

    public String getSize() {
        return size;
    }

    public void setSize(String size) {
        this.size = size;
    }


}
